# Stock Market Aalysis Using Time series

### Stocks are of IBM company using the Alphavantage

In [12]:
# Importing libraries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [13]:
# Reading the dataset
stocks_df = pd.read_excel("stocks.xlsx")
stocks_df.head()

date  1. open  2. high  3. low  4. close  5. volume
0  2021-06-24   145.80   146.19  144.69    145.44    3597137
1  2021-06-23   146.43   146.50  144.57    144.61    3202644
2  2021-06-22   146.52   146.81  145.35    146.36    2437121
3  2021-06-21   144.11   147.07  144.06    146.65    4285711
4  2021-06-18   144.48   144.68  143.04    143.12    9156505

In [14]:
stocks_df.shape

(5447, 6)

### The dataset contains 5447 records now lets us take all the records of one year till now

In [20]:
stocks_df[stocks_df["date"] == "2020-07-01"]

date  1. open  2. high  3. low  4. close  5. volume
247  2020-07-01   120.27   121.23  118.37    118.54    4657168

In [22]:
df = stocks_df.iloc[:248, ]

In [25]:
df = df.rename(columns={'1. open': "open", '2. high':"high", '3. low': "low", '4. close': "close", '5. volume': "volume"})

### Now we have to deal with data different from other data. As in stock prices we will only be given date and we have to predict all the other ourselves.